NationStatesAPIBot/NationStatesAPIBot/Manager/PermissionManager.cs

```diff
@@ -38,7 +38,14 @@ public Task RemoveRoleAsync(Role role, BotDbContext dbContext)
 
         public Task RevokePermissionAsync(string discordUserId, Permission permission, BotDbContext dbContext)
         {
-            throw new NotImplementedException();
+            return Task.Run(() =>
+            {
+                var user = dbContext.Users.Single(u => u.DiscordUserId == discordUserId);
+                var perm = user.UserPermissions.Find(p => p.Permission.Id == permission.Id);
+                var update = dbContext.Users.Update(user);
+                update.Entity.UserPermissions.Remove(perm);
+                dbContext.SaveChanges();
```

Comment:
You could use SaveChangesAsync() then you can await that resulting in Task.Run() not beeing necessary.

Reply:
That does make it much neater :)